### PR TITLE
Significant Change for Demo Day

### PR DESCRIPTION
**Pull request by PrismaPhonic**
This PR makes large architectural changes, schema changes, and provides new features to goodest doggo.  Some things that were done:

1. Better handling of mysql errors from mysql client library that we're using.
2. Refactor of business logic into query and command handlers to trim down the web api as much as possible.
3. A workspace where all relevant crates for the project can live.  This gets treated by all IDEs w/ rust plugins as a single unified project.
4. A custom pool built inside the infra crate using lazy_static.  This allows us to use the newest pooling available in the MySQL client library (currently rocket uses mysql 14.1 w/ an old version of r2d2, so this is much better.  It also moves all database logic out of the web api layer)
5. Retry wrapper for our query handler that will automatically retry up to 5 times.  Retry amount can be adjusted by changing the global constant.
6. Schema changes to have pupper id be a primary key, and to store ratings based on the pupper id.  vindex was changed from name to pupper id as well, and indexed pupper_id in the ratings table since it's most frequently in the WHERE clause.
7. A top 10 puppers page.  The way that we get the top 10 puppers is fairly naive right now (we get the top 10 by average rating, without weighting the amount of votes).  The process is that we query for the top 10 off the ratings table, and those go into an ordered hashmap, where the key is the puppers id, and the value is a `Pupper` struct instance.  Then we do a single query for all the winning puppers metadata using WHERE IN, and for each row result mutating the pupper at each key along the way, and finally turning the collection into a vector so the frontend can display a properly sorted top 10 list.  This was done this way because vitess does not currently support ORDER BY FIELD(), or ORDER BY FIND_IN_SET().
8. A new docker build image that uses the musl target, and the final docker build is only 15 mb (yay!)
9.  Probably a bunch of other cool stuff I'm forgetting.

### CODE REVIEW

File: deploy/goodestdoggo.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+# Orchestrator service
```
Comment (dkhenry) [inline deploy/goodestdoggo.yaml]: This isn't the orchestrator service

Comment (PrismaPhonic) [reply deploy/goodestdoggo.yaml]: This was a direct copy over from the multicell demo and I didn't change the comment.  Will remove

Comment (PrismaPhonic) [reply deploy/goodestdoggo.yaml]: Removed the outdated comment https://github.com/planetscale/goodest-doggo/pull/2/commits/7aeedffbffa2a60ba70223b23d4dff6db3f54a84

Comment (PrismaPhonic) [reply deploy/goodestdoggo.yaml]: PTAL

**Comment by dkhenry** [comment]
So more of a architecture question. I don't like how the queries are defined in doggo-core, but the actual query is in doggo-infra, and then to use it you need to handle it in doggo-api. If we could remove one of those three that would be better, if we could make it so you only needed to modify one package and the other two would be able to auto-wire up that would be even better.

**Comment by PrismaPhonic** [comment]
I think in a full fledged application core would be more extensive and handle validation, as well as enforcing complex business logic. Right now it's a bit funny being so thin.  I suppose for the time being we could get rid of core since no actual validation happens.  In a more extensive application I would put repositories in infra rather than query and command handlers.  The handlers would be in core where they would use repository traits, and otherwise handle validation and other complex business logic.  They would get wired up in api, and api would stay as thin as possible.  That way core is ignorant of infra (since it depends on repo traits), and infra simply is concrete implementation of those repo traits. 

In this case we are just demo'ing Vitess so the only benefit we would have by adding repos in infra would be to detach ourselves somewhat from the dependency on the mysql client library.  Maybe we would want to switch to Diesel in the future and that would be helpful, but right now Diesel doesn't work with Vitess because of session vars.  So I decided to keep it simple and wanted to have a package that could be built out for enforcing core business logic, which is core.  It's probably a bit too much "thinking ahead" and doesn't come across right now as having a strong purpose at the moment.

**Pull request merged**